src/goalServices/changeGoalStatus.ts

```diff
@@ -3,49 +3,51 @@ import db from '../models';
 
 interface GoalStatusChangeParams {
   goalId: number;
-  userId: number;
+  userId: number; // Always required, unless it's a system change.
   newStatus: string;
   reason: string;
   context: string;
   transaction?: Sequelize.Transaction;
+  oldStatusToUse?: string | null;
 }
 
 export default async function changeGoalStatus({
   goalId,
-  userId = 1,
+  userId = null,
```

Comment:
if we are counting on a user here, do we just want to error at the top of this function if there is no user ID? I agree that I don't think 1 makes sense as a default value but the rest of these changes seem to allow a goal status change without a user, unless I'm misunderstanding them.

Reply:
I had rewritten it to require the userId, and expected -1 to be passed when we are having the system instead of the user changing the status. But it is being reverted without providing any alternate method of setting the value to system.